src/replication/ng/adapter/native/adapter.go

```diff
@@ -0,0 +1,68 @@
+package native
+
+import (
+	"github.com/goharbor/harbor/src/common/utils/log"
+	adp "github.com/goharbor/harbor/src/replication/ng/adapter"
+	"github.com/goharbor/harbor/src/replication/ng/model"
+)
+
+const registryTypeNative model.RegistryType = "native"
+
+func init() {
+	if err := adp.RegisterFactory(registryTypeNative, func(registry *model.Registry) (adp.Adapter, error) {
+		return &native{
+			registry:             registry,
+			DefaultImageRegistry: adp.NewDefaultImageRegistry(registry),
+		}, nil
+	}); err != nil {
+		log.Errorf("failed to register factory for %s: %v", registryTypeNative, err)
+		return
+	}
+	log.Infof("the factory for adapter %s registered", registryTypeNative)
+}
+
+type native struct {
+	*adp.DefaultImageRegistry
+	registry *model.Registry
+}
+
+var _ adp.Adapter = native{}
+
+func (native) Info() (info *model.RegistryInfo, err error) {
+	return &model.RegistryInfo{
+		Type: registryTypeNative,
+		SupportedResourceTypes: []model.ResourceType{
+			model.ResourceTypeRepository,
+		},
+		SupportedResourceFilters: []*model.FilterStyle{
+			{
+				Type:  model.FilterTypeName,
+				Style: model.FilterStyleTypeText,
+			},
+			{
+				Type:  model.FilterTypeTag,
+				Style: model.FilterStyleTypeText,
+			},
+		},
+		SupportedTriggers: []model.TriggerType{
+			model.TriggerTypeManual,
+			model.TriggerTypeScheduled,
+			model.TriggerTypeEventBased,
```

Comment:
Docker registry shouldn't support the event based trigger.

Reply:
removed